### PR TITLE
Bootstrap with oracle invitations

### LINKED ISSUE #5242: Create fake Oracle for economic testing
## Problem to Solve

For economic stress test and other test deployments, we must enable directly specify the price, and have it propagated through.

## Design

Clone `ManualPriceAuthority` and move the `setPrice` operation onto the creator facet rather than the priceAuthority. Note that ManualPriceAuthority` is made to simplify unit tests. Most code is reused from elsewhere, so I don't think it's worth trying to make a single notifier that covers both use cases.

### PR DESCRIPTION
**Pull request by michaelfig**
closes: #5242 

## Description

Enough mechanism to get an oracle invite in the econ committee member's wallet on boot with `decentral-economy-config.json`.

image

### CODE REVIEW

**Comment by dckc** [comment]
> closes: https://github.com/Agoric/agoric-sdk/issues/5242

I'd like us to be more explicit about that. How do we manipulate the prices? Is an integration test feasible?

File: packages/run-protocol/src/proposals/committee-proposal.js [hunk 1]
```diff
@@ -1,46 +1,7 @@
...
-  };
-
-  return Promise.all(
-    paths.map(async path => {
-      assert(Array.isArray(path), X`path ${path} is not an array`);
-      return nextPath(nameAdmin, path);
-    }),
-  );
-};
+import { reserveThenDeposit } from './utils.js';
```
Comment (dckc) [inline packages/run-protocol/src/proposals/committee-proposal.js:4]: how about `nameHubAux.js` or `proposalUtils.js`? `utils.js` says very little about what's inside.

File: packages/vats/src/nameHub.js [hunk 2]
```diff
@@ -74,6 +74,28 @@ export const makeNameHubKit = () => {
         }
       }
     },
+    default(key, newValue, adminValue) {
```
Comment (dckc) [inline packages/vats/src/nameHub.js]: unit test, please?

File: packages/vats/src/nameHub.js [hunk 3]
```diff
@@ -74,6 +74,28 @@ export const makeNameHubKit = () => {
...
+      if (keyToRecord.has(key)) {
+        const record = keyToRecord.get(key);
+        if (!record.promise) {
+          // Already initalized.
+          return /** @type {any} */ (record.value);
+        }
+      }
+      return nameAdmin.update(key, newValue, adminValue);
+    },
+    set(key, newValue, adminValue) {
```
Comment (dckc) [inline packages/vats/src/nameHub.js]: unit test, please?

File: packages/run-protocol/scripts/price-feed-core.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  const proposalBuilder = powers =>
+    defaultProposalBuilder(powers, {
+      AGORIC_INSTANCE_NAME,
+      IN_BRAND_DECIMALS: parseInt(IN_BRAND_DECIMALS, 10),
+      OUT_BRAND_DECIMALS: parseInt(OUT_BRAND_DECIMALS, 10),
+      oracleAddresses,
+      brandIn: lookup(inLookup).catch(() => undefined),
+      brandOut: lookup(outLookup).catch(() => undefined),
+    });
+  await writeCoreProposal('gov-price-feed', proposalBuilder); // gov-price-feed.js gov-price-feed-permit.json
```
Comment (dckc) [inline packages/run-protocol/scripts/price-feed-core.js:100]: Managing names such as `'gov-price-feed'` seems awkward. That is: the fact that running this script creates a file with this prefix in the current directory, and we have to be sure they're distinct for all these scripts. hm.

**Review by dckc — COMMENTED** [review]
got called away; sharing what I've got so far...

File: packages/run-protocol/src/proposals/committee-proposal.js (hunk 1 above)
Comment (michaelfig) [reply packages/run-protocol/src/proposals/committee-proposal.js:4]: It's already `proposals/utils.js`.  Is that enough, or would you prefer `proposals/proposalUtils.js`?

File: packages/vats/src/nameHub.js (hunk 2 above)
Comment (michaelfig) [reply packages/vats/src/nameHub.js]: Gladly!  I was getting to that today.

File: packages/vats/src/nameHub.js (hunk 3 above)
Comment (michaelfig) [reply packages/vats/src/nameHub.js]: Also gladly!

File: packages/run-protocol/scripts/price-feed-core.js (hunk 4 above)
Comment (michaelfig) [reply packages/run-protocol/scripts/price-feed-core.js:100]: If we want better coordination, that seems like something for `agoric deploy ...` to endow.

File: packages/run-protocol/src/proposals/committee-proposal.js (hunk 1 above)
Comment (dckc) [reply packages/run-protocol/src/proposals/committee-proposal.js:4]: > Is that enough

ok

**Comment by michaelfig** [comment]
> I'd like us to be more explicit about that. How do we manipulate the prices? Is an integration test feasible?

I put some usage notes at the top of `manual-price-feed.js`.  I will think about how to do an integration test.

File: packages/vats/src/nameHub.js [hunk 5]
```diff
@@ -35,7 +37,7 @@ export const makeNameHubKit = () => {
       }
       return E(firstValue).lookup(...remaining);
     },
-    entries() {
+    entries: () => {
```
Comment (turadg) [inline packages/vats/src/nameHub.js:40]: we could fix this in one PR over the codebase, which might help the git blame history be more clear.

Comment (turadg) [inline packages/vats/src/nameHub.js:40]: we could fix this in one PR over the codebase, which might help the git blame history be more clear.

File: packages/vats/src/types.js [hunk 6]
```diff
@@ -38,11 +38,17 @@
...
  * deleted).
+ * @property {( key: string, newValue: unknown, newAdmin?: unknown) =>
+ *   Promise<any> } default Update if not already updated.  Return
+ *   existing value, or newValue if not existing.
  * @property {(
- *   key: string, newValue: unknown, newAdmin?: unknown) => void
+ *   key: string, newValue: unknown, newAdmin?: unknown) => Promise<any>
+ * } set Update only if already initialized. Reject if not.
+ * @property {(
+ *   key: string, newValue: unknown, newAdmin?: unknown) => Promise<any>
```
Comment (turadg) [inline packages/vats/src/types.js]: Why return `newValue`?

As above:
```suggestion
 * @property {<T>(
 *   key: string, newValue: T, newAdmin?: unknown) => Promise<T>
```

File: packages/vats/src/types.js [hunk 7]
```diff
@@ -38,11 +38,17 @@
  * @property {(key: string) => void} reserve Mark a key as reserved; will
  * return a promise that is fulfilled when the key is updated (or rejected when
  * deleted).
+ * @property {( key: string, newValue: unknown, newAdmin?: unknown) =>
+ *   Promise<any> } default Update if not already updated.  Return
```
Comment (turadg) [inline packages/vats/src/types.js:42]: Can we assume the new value is the same type as the prior value?
```suggestion
 * @property {<T>( key: string, newValue: T, newAdmin?: unknown) =>
 *   Promise<T> } default Update if not already updated.  Return
```

File: packages/vats/src/types.js [hunk 8]
```diff
@@ -38,11 +38,17 @@
...
+ *   existing value, or newValue if not existing.
  * @property {(
- *   key: string, newValue: unknown, newAdmin?: unknown) => void
+ *   key: string, newValue: unknown, newAdmin?: unknown) => Promise<any>
+ * } set Update only if already initialized. Reject if not.
```
Comment (turadg) [inline packages/vats/src/types.js:46]: I don't see this used anywhere. Are we going to need it?

**Review by dckc — APPROVED** [review]
LGTM.

File: packages/vats/src/nameHub.js (hunk 5 above)
Comment (michaelfig) [reply packages/vats/src/nameHub.js:40]: I prefer to do it incrementally, but I understand your point.

File: packages/vats/src/types.js (hunk 7 above)
Comment (michaelfig) [reply packages/vats/src/types.js:42]: No, we can't make that assumption.

File: packages/vats/src/types.js (hunk 8 above)
Comment (michaelfig) [reply packages/vats/src/types.js:46]: I put it in for symmetry with the stores.

File: packages/vats/src/types.js (hunk 6 above)
Comment (michaelfig) [reply packages/vats/src/types.js]: Okay, I won't return newValue.

Comment (turadg) [reply packages/vats/src/types.js]: Looks like you made `set` and `update` return void but didn't change the runtime. 

https://github.com/Agoric/agoric-sdk/commit/ac12696a10372d66cff048afd78d6750d9771011#diff-42f4291b619264addc7a7e12a8902cf3c60e5f797e97b5e4fa2a1e0c01d111ffR101

https://github.com/Agoric/agoric-sdk/commit/ac12696a10372d66cff048afd78d6750d9771011#diff-42f4291b619264addc7a7e12a8902cf3c60e5f797e97b5e4fa2a1e0c01d111ffR122

Comment (michaelfig) [reply packages/vats/src/types.js]: Oops.  I'll fix that right away!

**Pull request merged**